Add deprecation note to HCP Vagrant related pages (#3138)
hashicorp/web-unified-docs: content/global/partials/vagrant-eol.mdx

<Note>

> HCP Vagrant is being <a href="/hcp/docs/vagrant/hcp-vagrant-eol">deprecated</a>.  Customers using Vagrant boxes hosted on the HCP Vagrant registry to find a new provider and rehost their boxes.  New boxes can no longer be created in HCP Vagrant as of December 14, 2026.  Existing box deployments will continue to operate until the end of support date of March 15, 2027.  After that time, all deployments will be decommissioned.

Reviewer: ```suggestion
HCP Vagrant is being [deprecated](/hcp/docs/vagrant/hcp-vagrant-eol). Customers using Vagrant boxes hosted on the HCP Vagrant registry must find a new provider and re-host their boxes. As of December 14, 2026, you can no longer create new boxes in HCP Vagrant. Existing box deployments will continue to operate until the end of support date of March 15, 2027.  After that time, we will decommission all deployments.
```
HTLM => Markdown link
Switch to active voice 
I think there's a typo in the second line? Sometimes I can't read. 
That third sentence still sounds weird to me because of the present tense against the future event, so use whichever sounds best to you
